VERIFY3(dev->l2ad_hand + distance < dev->l2ad_end) failed

<!-- Please fill out the following template, which will help other contributors address your issue. -->

<!--
Thank you for reporting an issue.

*IMPORTANT* - Please check our issue tracker before opening a new issue.
Additional valuable information can be found in the OpenZFS documentation
and mailing list archives.

Please fill in as much of the template as possible.
-->

### System information
<!--  add version after "|" character -->
Type | Version/Name
 --- | ---
Distribution Name	| FreeBSD
Distribution Version	| 13.3-STABLE
Kernel Version	| 13.3-STABLE
Architecture	| amd64
OpenZFS Version	| zfs-2.1.15-FreeBSD_gfb6d53206
<!--
Command to find OpenZFS version:
zfs version

Commands to find kernel version:
uname -r		# Linux
freebsd-version -r	# FreeBSD
-->

### Describe the problem you're observing

While receiving an rsync incremental update, and doing nothing else, my server crashed with

panic: VERIFY3(dev->l2ad_hand + distance < dev->l2ad_end) failed (161060749312 < 161060749312)

This is in `l2arc_evict` in arc.c.  I suspect `<` may need to be `<=`.

### Describe how to reproduce the problem

It happened only once.

### Include any warning/errors/backtraces from the system logs
<!--
*IMPORTANT* - Please mark logs and text output from terminal commands
or else Github will not display them correctly.
An example is provided below.

Example:
```
this is an example how log text should be marked (wrap it with ```)
```
-->

The `dev` pointer points at
```
 l2ad_hand = 161051246592,
 l2ad_start = 4198400,
 l2ad_end = 161060749312,
 l2ad_first = 0,
 l2ad_writing = 0,
```
Variable `distance` is 9502720, equal to `l2ad_end - l2ad_hand`, as expected from the panic message.

My server has a raidz2 pool on hard drives with a cache on SSD.  RAM is 24 GB.

Reported to FreeBSD at https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=279038.

--- a/module/zfs/arc.c
+++ b/module/zfs/arc.c
@@ -8886,7 +8886,7 @@ out:
 		 * assertions may be violated without functional consequences
 		 * as the device is about to be removed.
 		 */
-		ASSERT3U(dev->l2ad_hand + distance, <, dev->l2ad_end);
+		ASSERT3U(dev->l2ad_hand + distance, <=, dev->l2ad_end);
 		if (!dev->l2ad_first)
 			ASSERT3U(dev->l2ad_hand, <=, dev->l2ad_evict);
 	}

